### DISCUSSION TITLE
Progressive migration from redux: HOC

### DISCUSSION THREAD

**Discussion by schontz** [mobxjs/mobx#3539] [Q&A] [answered]
We are looking at migrating a redux/saga codebase to mobx, for obvious reasons. Since our codebase is quite large, we would like to do things progressively, which means redux and mobx will be running together. We have function and class components.

Are there any notes for this kind of set up? For example, when hooking up to components, which HOC comes first?

```javascript
// before: just redux 😔
export default connect(...)(SomeComponent);

// both: mobx + redux 🤨
export default connect(...)(observer(SomeComponent));
// or
export default observer(connect(..)(SomeComponent));

// future: just mobx 😍
export default observer(SomeComponent);
```

**Comment by urugator** [comment]
`observer(connect(..)(SomeComponent));` won't probably work in a sense that `SomeComponent` won't track accessed observables.

`connect(...)(observer(SomeComponent));` should work, but you must not access any observables in `mapStateToProps` (it doesn't run as part of the observer). You can only pass them through.

You may actually try `observer(connect(...)(observer(SomeComponent)))`.

**Comment by mweststrate** [comment, marked as answer]
The first
